Hi again, @alessandromontanari !

On my side, a local test is failing, due to geometry issue (some modules off for the run 7213):

```
$ python start_dqm.py -c NectarCAMQM --runnb 7213 --max-events 3000 --plot $NECTARCAMDATA $NECTARCAMDATA
[...]
Traceback (most recent call last):
  File "/home/jlenain/local/src/python/cta-observatory/nectarchain/src/nectarchain/dqm/start_dqm.py", line 341, in <module>
    main()
  File "/home/jlenain/local/src/python/cta-observatory/nectarchain/src/nectarchain/dqm/start_dqm.py", line 321, in main
    fig_dict, name_dict = p.plot_results(name, fig_path)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jlenain/local/src/python/cta-observatory/nectarchain/src/nectarchain/dqm/charge_integration.py", line 266, in plot_results
    self._plot_camera_image(image, title, text, filename, key, fig_path)
  File "/home/jlenain/local/src/python/cta-observatory/nectarchain/src/nectarchain/dqm/charge_integration.py", line 248, in _plot_camera_image
    disp.image = image
    ^^^^^^^^^^
  File "/home/jlenain/local/opt/conda/envs/nectar-dev/lib/python3.11/site-packages/ctapipe/visualization/mpl_camera.py", line 369, in image
    raise ValueError(
ValueError: Image has a different shape (1855,) than the given CameraGeometry (1841,)
```

--- a/src/nectarchain/dqm/camera_monitoring.py
+++ b/src/nectarchain/dqm/camera_monitoring.py
@@ -6,6 +6,7 @@ import numpy as np
 from astropy import time as astropytime
 from ctapipe.coordinates import EngineeringCameraFrame
 from ctapipe.visualization import CameraDisplay
+from ctapipe_io_nectarcam import constants
 from matplotlib import pyplot as plt
 
 from .dqm_summary_processor import DQMSummary
@@ -113,7 +114,6 @@ class CameraMonitoring(DQMSummary):
 
             TotalDrawers = int(np.max(self.DrawerNum2))
             n_drawers = TotalDrawers + 1
-            PIXELS_PER_DRAWER = 7  # NectarCAM has 7 pixels per drawer
 
             drawer_temp1_mean = np.zeros(n_drawers)
             drawer_temp2_mean = np.zeros(n_drawers)
@@ -136,10 +136,19 @@ class CameraMonitoring(DQMSummary):
                     drawer_temp2_mean[i] = np.nan
                     drawer_temp2_std[i] = np.nan
 
-            self.DrawerTemp1_mean = np.repeat(drawer_temp1_mean, PIXELS_PER_DRAWER)
-            self.DrawerTemp2_mean = np.repeat(drawer_temp2_mean, PIXELS_PER_DRAWER)
-            self.DrawerTemp1_std = np.repeat(drawer_temp1_std, PIXELS_PER_DRAWER)
-            self.DrawerTemp2_std = np.repeat(drawer_temp2_std, PIXELS_PER_DRAWER)
+            # NectarCAM has 7 pixels per drawer
+            self.DrawerTemp1_mean = np.repeat(
+                drawer_temp1_mean, constants.N_PIXELS_MODULE
+            )
+            self.DrawerTemp2_mean = np.repeat(
+                drawer_temp2_mean, constants.N_PIXELS_MODULE
+            )
+            self.DrawerTemp1_std = np.repeat(
+                drawer_temp1_std, constants.N_PIXELS_MODULE
+            )
+            self.DrawerTemp2_std = np.repeat(
+                drawer_temp2_std, constants.N_PIXELS_MODULE
+            )
 
             self.DrawerTemp1_trend = np.array(
                 [self.DrawerTemp12[self.DrawerNum2 == ii] for ii in range(n_drawers)]

--- a/src/nectarchain/dqm/charge_integration.py
+++ b/src/nectarchain/dqm/charge_integration.py
@@ -146,6 +146,10 @@ class ChargeIntegrationHighLowGain(DQMSummary):
 
         image = output[0]
         peakpos = output[1]
+        if np.any(np.isnan(image) | np.isinf(image)):
+            image[np.isnan(image) | np.isinf(image)] = np.zeros_like(
+                image[np.isnan(image) | np.isinf(image)]
+            )
 
         if evt.trigger.event_type == EventType.SKY_PEDESTAL:
             # count sky peds, event id 2
@@ -244,7 +248,8 @@ class ChargeIntegrationHighLowGain(DQMSummary):
 
     def _plot_camera_image(self, image, title, text, filename, key, fig_path):
         fig, disp = plt.subplots()
-        disp = CameraDisplay(geometry=self.camera[~self.pixelBADplot[0]])
+        image[self.pixelBADplot[0]] = 0
+        disp = CameraDisplay(geometry=self.camera)
         disp.image = image
         disp.cmap = plt.cm.coolwarm
         disp.axes.text(2, -0.8, text, fontsize=12, rotation=90)
@@ -318,22 +323,30 @@ class ChargeIntegrationHighLowGain(DQMSummary):
             key = f"CHARGE-INTEGRATION-IMAGE-PED-RMS-{self.gain_c.upper()}-GAIN"
             self._plot_camera_image(image, title, text, filename, key, fig_path)
 
+        indexes_bad_pixels = np.where(self.pixelBADplot[0] == True)[0]
+
         # Charge integration SPECTRUM
         if self.counter_evt > 0:
             fig, _ = plt.subplots()
             for i in range(len(self.pixels)):
-                plt.hist(
-                    self.image_all[:, i],
-                    100,
-                    fill=False,
-                    density=True,
-                    stacked=True,
-                    linewidth=1,
-                    log=True,
-                    alpha=0.01,
-                )
+                if i not in indexes_bad_pixels:
+                    plt.hist(
+                        self.image_all[:, i],
+                        100,
+                        fill=False,
+                        density=True,
+                        stacked=True,
+                        linewidth=1,
+                        log=True,
+                        alpha=0.01,
+                    )
             plt.hist(
-                np.mean(self.image_all, axis=1),
+                np.mean(
+                    self.image_all[
+                        :, ~np.isin(np.arange(len(self.pixels)), indexes_bad_pixels)
+                    ],
+                    axis=1,
+                ),
                 100,
                 color="r",
                 linewidth=1,
@@ -360,18 +373,24 @@ class ChargeIntegrationHighLowGain(DQMSummary):
         if self.counter_ped > 0:
             fig, _ = plt.subplots()
             for i in range(len(self.pixels)):
-                plt.hist(
-                    self.image_ped[:, i],
-                    100,
-                    fill=False,
-                    density=True,
-                    stacked=True,
-                    linewidth=1,
-                    log=True,
-                    alpha=0.01,
-                )
+                if i not in indexes_bad_pixels:
+                    plt.hist(
+                        self.image_ped[:, i],
+                        100,
+                        fill=False,
+                        density=True,
+                        stacked=True,
+                        linewidth=1,
+                        log=True,
+                        alpha=0.01,
+                    )
             plt.hist(
-                np.mean(self.image_ped, axis=1),
+                np.mean(
+                    self.image_ped[
+                        :, ~np.isin(np.arange(len(self.pixels)), indexes_bad_pixels)
+                    ],
+                    axis=1,
+                ),
                 100,
                 color="r",
                 linewidth=1,

--- a/src/nectarchain/dqm/start_dqm.py
+++ b/src/nectarchain/dqm/start_dqm.py
@@ -314,6 +314,7 @@ def main():
         else:
             db.abort_and_close()
 
+    start_plotting = time.time()
     # if plot option in arguments, it will construct the figures and save them
     if PlotFig:
         figures_to_save = []
@@ -325,6 +326,7 @@ def main():
         for fig, path in figures_to_save:
             fig.savefig(path)
             plt.close(fig)
+    plotting_time = time.time() - start_plotting
 
     end = time.time()
     log.info(
@@ -332,7 +334,8 @@ def main():
         + f"Process: {process_time:.2f}s, "
         + f"Finish: {finish_time:.2f}s, "
         + f"Results: {results_time:.2f}s, "
-        + f"Write: {write_time:.2f}s"
+        + f"Write: {write_time:.2f}s, "
+        + f"Plotting: {plotting_time:.2f}s"
     )
     log.info(f"Processing time: {end-start:.2f} s.")
 

